pkg/cloudprovider/cloudprovider.go

```diff
@@ -104,7 +104,11 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 	if len(instanceTypes) == 0 {
 		return nil, cloudprovider.NewInsufficientCapacityError(fmt.Errorf("all requested instance types were unavailable during launch"))
 	}
-	instance, err := c.instanceProvider.Create(ctx, nodeClass, nodeClaim, getTags(ctx, nodeClass, nodeClaim), instanceTypes)
+	tags, err := getTags(ctx, nodeClass, nodeClaim)
+	if err != nil {
+		return nil, cloudprovider.NewCreateError(fmt.Errorf("tag validation bypassed, %w", err), "Error tag validation bypassed")
```

Comment:
Rather than a CreateError, I think this should be a NodeClassNotReady error so that the NodeClaim gets cleaned up and we retry with another NodeClass

Reply:
Done.